tests/testthat/test-utils.R

```diff
@@ -122,9 +121,51 @@ test_that("strict_mode_error() works correctly [BBR-UTL-010]", {
   })
 })
 
-
 test_that("suppressSpecificWarning() works [BBR-UTL-011]", {
   # log() of a negative number raises a warning
   x <- suppressSpecificWarning(log(-1), "NaNs produced")
   expect_true(is.nan(x))
 })
+
+test_that("warning raised when threads > 1 and parallel is FALSE [BBR-UTL-014]", {
+  withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()),{
+    withr::with_tempdir({
+      fs::dir_create(file.path(tempdir(), "test_path"))
+
+      on.exit(if(fs::dir_exists(file.path(tempdir(),"test_path")))
+      {
+        fs::dir_delete(file.path(tempdir(),"test_path"))
+      })
+
+      files_to_copy <- file.path(ABS_MODEL_DIR, c("1.ctl"))
+
+      fs::file_copy(system.file("extdata", "acop.csv", package = "bbr"), file.path(tempdir(), "test_path"))
+      purrr::walk(files_to_copy, fs::file_copy, file.path(tempdir(), "test_path"))
+      fs::dir_copy(file.path(ABS_MODEL_DIR, "1"), file.path(tempdir(), "test_path"))
+      ctl <- read_lines(file.path(tempdir(),"test_path", "1.ctl")) %>%  stringr::str_remove("../../../../extdata/")
+      write_lines(ctl, file.path(tempdir(),"test_path", "1.ctl"))
+
+      mod1 <- new_model(file.path(tempdir(), "test_path", "1"), .description = "original test-workflow-bbi model",
+                        .tags = ORIG_TAGS,.bbi_args = list(overwrite = TRUE, threads = 2))
+
+      readr::write_file("created_by: test-utils", file.path(tempdir(), "test_path", "bbi.yaml"))
+
+
+      res <- capture.output(submit_model(mod1, .dry_run = TRUE))
+
+      #Testing that check_bbi_args is appending --parallel when not passed
+      expect_identical("--parallel", str_subset(res,"--parallel") %>% str_extract("--parallel"))
+      fs::file_delete(file.path(tempdir(), "test_path", "1.yaml"))
+
+
+      mod1 <-  new_model(file.path(tempdir(), "test_path", "1"), .description = "original test-workflow-bbi model",
+                         .tags = ORIG_TAGS,.bbi_args = list(overwrite = TRUE, threads = 2, parallel = FALSE))
+
+      #When `threads` > 1 and parallel is False raise a warning
+      expect_warning(submit_model(mod1, .dry_run = TRUE))
```

Comment:
When using `expect_warning()`, we want to have a specific warning we're testing for. If the warning does not match, we want it to be spit back to us.

Something like:
```r
expect_warning(submit_model(mod1, .dry_run = TRUE), "`threads > 1` but model will not run in parallel because `parallel = FALSE`")
```

Though note: I sometimes have to play around with some regex or `expect_warning()` arguments to get them to say they actually match (when they do)